www/adad05/cv05/classes/Database.php

```diff
@@ -0,0 +1,13 @@
+<?php
+
+require_once 'DatabaseOperations.php';
+require_once 'DatabaseConnection.php';
+
+abstract class Database implements DatabaseOperations
+{
+    protected $pdo;
+    public function __construct()
+    {
+        $pdo = DatabaseConnection::getPDOConnection();

```

Comment:
tohle nebude fungovat ($pdo nebude dostupne v instancich podrazenych trida)

Reply:
Dobrý den, původně jsem tam měl pouze require, ale házelo mi to error. Poradil byste mi, jak na to správně?
Zároveň bych se rád zeptal, jestli máte radu jak vyřešit skrytí hesla od mysql databáze, aby nebylo explicitně v kódu.

Předem děkuji. Daniel Adam

Reply:
Dobrý den, bude řešením vytvoření metody pro vrácení atributu $pdo?
Zároveň bych se rád zeptal, jestli máte radu, jak skrýt heslo od mySQL databáze, aby nebylo explicitně v kódu.

Předem děkuji. Daniel Adam

Reply:
`__construct` se zavola s vytvarenim instanci, a reference na vytvorenou instanci je `$this`, takze `$this->pdo = ...`

heslo vymazat z kodu a pak teprve nahrat na git

Reply:
Děkuji. Nahrál jsem nový commit.